### DISCUSSION TITLE
Validation error while saving the form from custom page

### DISCUSSION THREAD

**Discussion by aimonext** [filamentphp/filament#8382] [Help] [answered]
I've crafted a custom page and established a form successfully. However, when I invoke the $this->validate() method within the save method, it results in an error.
# The error
![image](https://github.com/filamentphp/filament/assets/110348549/43231b3c-8a22-470c-91b0-e6f83afae2be)
# Code
```php
class Security extends Page implements HasForms
{
    use InteractsWithFormActions, InteractsWithForms;

    protected static bool $shouldRegisterNavigation = false;

    protected static string $view = 'filament.pages.security';

    protected function getHeaderActions(): array
    {
        return [
            Action::make('go_to_profile')
                ->icon('heroicon-m-pencil-square')
                ->url(fn () => route('filament.user.auth.profile')),
        ];
    }

    public function form(Form $form): Form
    {
        return $form->schema([
            TextInput::make('old_password')->required()->password(),
            TextInput::make('new_password')->required()->password()->confirmed(),
            TextInput::make('new_password_confirmation')->required()->password()->same("new_password"),
        ]);
    }

    public function getFormActions(): array
    {
        return [
            Action::make('change_password')->submit(),
        ];
    }

    public function save()
    {
        $this->validate(['old_password' => 'required']);
    }
}
```

**Comment by danharrin** [comment, marked as answer]
Follow the "adding a form to a Livewire component" docs.

1) You have no `$this->form->fill()` in `mount()`
2) You are manually validating when there is no need, you should use `$this->form->getState()`
3) The error is because you have no `statePath('data')` and `public $data` OR individual properties for each field.

**Reply by aimonext** [reply]
thanks for your response

**Reply by aimonext** [reply]
@danharrin, I'm encountering another issue. When I attempt to change the password, it unexpectedly redirects me to '/login,' implying that the user is logged out. Strangely, there's a peculiar error that appears initially, as shown below:
![image](https://github.com/filamentphp/filament/assets/110348549/91862c54-ed60-4594-a5b4-9abd07bd6a4a)
### However, on the second reload, it correctly redirects to the intended destination.

**Reply by danharrin** [reply]
Laravel bug, set up a redirect from /laravel/login to your admin panel login page, and give it `name('login')`

**Reply by aimonext** [reply]
thanks
